modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -576,6 +580,27 @@
   [driver ps i t]
   (sql-jdbc.execute/set-parameter driver ps i (t/sql-timestamp (t/with-zone-same-instant t (t/zone-id "UTC")))))
 
+;;; --------------------------------------------------- Query remarks ---------------------------------------------------
+
+;  Snowflake strips comments prepended to the SQL statement (default remark injection behavior). We should append the
+;  remark instead.
+(defmethod sql-jdbc.execute/inject-remark :snowflake
+  [_ sql remark]
+  (str sql "\n\n-- " remark))
+
+(defmethod qp.util/query->remark :snowflake
+  [_ {{:keys [executed-by card-id dashboard-id query-hash context]} :info, query-type :type, database-id :database}]
+  (json/generate-string {:client      "Metabase"
+                         :context     context
+                         :queryType   (case (keyword query-type) :query "MBQL" :native "native")
```

Comment:
Converting to keyword is not really necessary. I suggest adding a default so that it doesn't blow up if an unexpected `query-type` comes in.

Reply:
Done, thanks! I had copied and pasted that code from the query remark implementation from another driver, but now that I think about it, I don't see that much value in having a `case` to map `query` to `MBQL` and there's already a lot of difference in the existing query remarks format and content between different drivers, so I don't see any harm in having it different here. I updated the code to just use `query-type` directly. Let me know if you think it is not a good idea.

Reply:
I think that's OK, although the translation might be clearer to newcomers, because native queries are also queries. "MBQL" and "native" are definitely disjoint notions.